examples/common/chip-app-server/server.cpp

```diff
@@ -0,0 +1,96 @@
+#include "server.h"
+
+#include <string.h>
+
+#include <inet/IPAddress.h>
+#include <inet/InetError.h>
+#include <inet/InetLayer.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <support/CodeUtils.h>
+#include <support/ErrorStr.h>
+#include <support/logging/CHIPLogging.h>
+#include <sys/param.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/UDP.h>
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::Transport;
+using namespace ::chip::DeviceLayer;
+
+#ifndef EXAMPLE_SERVER_NODEID
+// "nRF5"
+#define EXAMPLE_SERVER_NODEID 12344321
+#endif // EXAMPLE_SERVER_NODEID
+
+// Transport Callbacks
+namespace {
+extern "C" void HandleDataModelMessage(const MessageHeader & header, System::PacketBuffer * buffer, SecureSessionMgrBase * mgr);
```

Comment:
I would vaguely prefer this declaration live in `DataModelHandler.h` and we include that here, but I guess that might need to wait until we have a single `HandleDataModelMessage` implementation across all servers?  That would also give us a place to hang API documentation (like what the callee does with the `PacketBuffer`'s refcount).

Reply:
I have some attempts and failed, currently HandleDataModelMessage includes `app/util/util.h` which includes files from `gen` dir, cause HandleDataModelMessage (as well as InitDataModel) need to stay with `gen` dir.